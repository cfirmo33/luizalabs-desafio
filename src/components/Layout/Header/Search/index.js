import React from 'react'
import Icon from '../../../Layout/Icon'
import PropTypes from 'prop-types'

const Search = ({ value, onChange }) => (
  <form className="search-bar" onSubmit={(event) => { event.preventDefault() }}>
    <Icon icon="fas fa-search" />
    <input
      type="text"
      value={value}
      onChange={onChange}
      placeholder="Qual o nome do personagem?"
    />
  </form>
)

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
}

export default Search
