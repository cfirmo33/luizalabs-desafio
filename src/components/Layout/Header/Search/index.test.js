import React from 'react'
import { shallow } from 'enzyme'
import Search from './index'

describe('Search', () => {
  it('Deve renderizar corretamente', () => {
    const value = ''
    const wrapper = shallow(
      <Search value={value} onChange={() => { }} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
