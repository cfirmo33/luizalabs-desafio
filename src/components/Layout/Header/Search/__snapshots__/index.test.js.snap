// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Deve renderizar corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    onChange={[Function]}
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          icon="fas fa-search"
/>,
        <input
          onChange={[Function]}
          placeholder="Qual o nome do personagem?"
          type="text"
          value=""
/>,
      ],
      "className": "search-bar",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "fas fa-search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Qual o nome do personagem?",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            icon="fas fa-search"
/>,
          <input
            onChange={[Function]}
            placeholder="Qual o nome do personagem?"
            type="text"
            value=""
/>,
        ],
        "className": "search-bar",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "fas fa-search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Qual o nome do personagem?",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
